src/diffusers/loaders.py

```diff
@@ -943,14 +943,19 @@ def _modify_text_encoder(self, attn_processors: Dict[str, LoRAAttnProcessor]):
                 module = self.text_encoder.get_submodule(name)
                 # Construct a new function that performs the LoRA merging. We will monkey patch
                 # this forward pass.
-                lora_layer = getattr(attn_processors[name], self._get_lora_layer_attribute(name))
+                attn_processor_name = ".".join(name.split(".")[:-1])
+                lora_layer = getattr(attn_processors[attn_processor_name], self._get_lora_layer_attribute(name))
                 old_forward = module.forward
 
-                def new_forward(x):
-                    return old_forward(x) + lora_layer(x)
+                # create a new scope that locks in the old_forward, lora_layer value for each new_forward function
+                def make_new_forward(old_forward, lora_layer):
```

Comment:
Here I would maybe add a reference to https://github.com/huggingface/diffusers/pull/3490#issuecomment-1555059060 for users to refer to.

Reply:
```suggestion
                # create a new scope that locks in the old_forward, lora_layer value for each new_forward function
                # for more detail, see https://github.com/huggingface/diffusers/pull/3490#issuecomment-1555059060
                def make_new_forward(old_forward, lora_layer):
```
Shall we add a link reference like this?

Reply:
Sure.